Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -341,51 +345,50 @@ PhysicalParticleContainer::AddPlasmaFromFile(const std::string s_f,
 #ifdef WARPX_USE_OPENPMD
     //TODO: Make changes for read/write in multiple MPI ranks
     if (ParallelDescriptor::IOProcessor()) {
-        openPMD::Series series = openPMD::Series(s_f,
-                                                 openPMD::AccessType::READ_ONLY);
-        amrex::Print() << "openPMD standard version " << series.openPMD() << "\n";
-        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(series.iterations.size() == 1u, "External "
-                                         "file should contain only 1 iteration\n");
-        openPMD::Iteration& i = series.iterations[1];
-        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(i.particles.size() == 1u, "External file "
-                                         "should contain only 1 species\n");
-        std::pair<std::string,openPMD::ParticleSpecies> ps = *i.particles.begin();
-
-        //TODO: Add ASSERT_WITH_MESSAGE to test if mass and charge are both const
-        amrex::ParticleReal p_m = ps.second["mass"][openPMD::RecordComponent::SCALAR].loadChunk<amrex::ParticleReal>().get()[0];
-        double const mass_unit = ps.second["mass"][openPMD::RecordComponent::SCALAR].unitSI();
-        amrex::ParticleReal p_q = ps.second["charge"][openPMD::RecordComponent::SCALAR].loadChunk<amrex::ParticleReal>().get()[0];
-        double const charge_unit = ps.second["charge"][openPMD::RecordComponent::SCALAR].unitSI();
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(plasma_injector,
+                                         "AddPlasmaFromFile: plasma injector not initialized.\n");
+        // take ownership of the series and close it when done
+        auto series = std::move(plasma_injector->m_openpmd_input_series);
```

Comment:
We intentionally:
```
// take ownership of the series and close it when done
```
:)

Reply:
Yeah, I did read that the first time, but I do not know what closing a series means, sorry :confused:. Is there a pointer `delete` operation at the end?

Reply:
`std::unique_ptrs` do `delete` the memory they own on destruction. they basically save us from ever writing `new` or `delete` again because their lifetime (or a `.reset()` call) determines the lifetime of the memory they own.